File: poc-kronos-frontend/src/model/disciplina/Disciplina.js
class Disciplina {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.codigo = dados ? dados.codigo : null;
    this.dataInicioVigencia = dados ? dados.dataInicioVigencia : null;
    this.dataFinalVigencia = dados ? dados.dataFinalVigencia : null;
    this.nome = dados ? dados.nome : null;
    this.sigla = dados ? dados.sigla : null;
    this.cargaHorariaTotal = dados ? dados.cargaHorariaTotal : null;
    this.cargaHorariaPresencial = dados ? dados.cargaHorariaPresencial : null;
    this.cargaHorariaDistancia = dados ? dados.cargaHorariaDistancia : null;
    this.ativa = dados ? dados.ativa : null;
    this.obrigatoria = dados ? dados.obrigatoria : null;
    this.valor = dados ? dados.valor : null;
    this.idTipoDisciplina = dados ? dados.idTipoDisciplina : null;
    this.nomeTipoDisciplina = dados ? dados.nomeTipoDisciplina : null;
    this.idTipoRegimeLetivo = dados ? dados.idTipoRegimeLetivo : null;
    this.nomeTipoRegimeLetivo = dados ? dados.nomeTipoRegimeLetivo : null;
    this.idPeriodo = dados ? dados.idPeriodo : null;
    this.siglaPeriodo = dados ? dados.siglaPeriodo : null;
    this.idComponente = dados ? dados.idComponente : null;
    this.preRequisitos = dados ? dados.preRequisitos : null;
    this.idDisciplinaUnificacao = dados ? dados.idDisciplinaUnificacao : null;
    this.siglaDisciplinaUnificacao = dados
      ? dados.siglaDisciplinaUnificacao
      : null;
  }

  static validations(required, maxLength, numeric, decimal, dataFimMenor) {
    return {
      codigo: { required, numeric },
      nome: { required, maxLength: maxLength(100) },
      sigla: { required, maxLength: maxLength(20) },
      cargaHorariaTotal: { required, numeric },
      cargaHorariaPresencial: { required, numeric },
      cargaHorariaDistancia: { required, numeric },
      dataInicioVigencia: { required },
      dataFinalVigencia: { dataFimMenor: dataFimMenor("dataInicioVigencia") },
      valor: { required, decimal },
      idTipoDisciplina: { required },
      idTipoRegimeLetivo: { required },
      idPeriodo: { required }
    };
  }

  static validationMessages() {
    return {
      nome: {
        required: "Nome ?? obrigat??rio",
        maxLength: "Nome s?? aceita no m??ximo 100 caracteres"
      },
      sigla: {
        required: "Sigla ?? obrigat??ria",
        maxLength: "Sigla s?? aceita no m??ximo 200 caracteres"
      },
      codigo: {
        required: "C??digo ?? obrigat??rio",
        numeric: "Somente n??meros"
      },
      dataInicioVigencia: {
        required: "Data Inicio Vig??ncia ?? obrigat??rio"
      },
      dataFinalVigencia: {
        dataFimMenor:
          "Data Final de Vig??ncia n??o pode ser menor que a Data In??cio de Vig??ncia"
      },
      cargaHorariaTotal: {
        required: "Carga Hor??ria Total ?? obrigat??ria",
        numeric: "Somente n??meros"
      },
      cargaHorariaDistancia: {
        required: "Carga Hor??ria Dist??ncia ?? obrigat??ria",
        numeric: "Somente n??meros"
      },
      cargaHorariaPresencial: {
        required: "Carga Hor??ria Presencial ?? obrigat??ria",
        numeric: "Somente n??meros"
      },
      valor: {
        required: "Valor ?? obrigat??rio",
        decimal: "Somente valor decimal"
      },
      idTipoDisciplina: {
        required: "Disciplina ?? obrigat??ria"
      },
      idTipoRegimeLetivo: {
        required: "Tipo Regime Letivo ?? obrigat??rio"
      },
      idPeriodo: {
        required: "Per??odo ?? obrigat??rio"
      }
    };
  }
}
export default Disciplina;

File: src/main/java/br/com/kronos/backend/aplicacao/disciplina/Disciplina.java
package br.com.kronos.backend.aplicacao.disciplina;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Disciplina {
	
	public static final String COMBO_CACHE_NAME_PERFIL_PROFESSORES = "combo_disciplinas_perfil_professores";
	public static final String COMBO_CACHE_NAME_DISCIPLINA_UNIFICA = "combo_disciplinas_unificacoes";
    
    private long id;
    private Integer codigo;
    private String nome;
    private String sigla;
    private Integer cargaHorariaTotal;
    private Integer cargaHorariaPresencial;
    private Integer cargaHorariaDistancia;
    private boolean ativa;
    private boolean obrigatoria;
    private Double valor;
    private Integer idTipoDisciplina;
    private Integer idTipoRegimeLetivo;
    private Long idPeriodo;
    private Long idComponente;
    private LocalDate dataInicioVigencia;
    private LocalDate dataFinalVigencia;
    private boolean preRequisitos;
    private Long idDisciplinaUnifica;

	@Builder
	public Disciplina(long id, Integer codigo, String nome, String sigla, Integer cargaHorariaTotal, Integer cargaHorariaPresencial, Integer cargaHorariaDistancia, 
    boolean ativa, boolean obrigatoria, Double valor, Integer idTipoDisciplina, Integer idTipoRegimeLetivo, Long idPeriodo, 
    Long idComponente, LocalDate dataInicioVigencia, LocalDate dataFinalVigencia, boolean preRequisitos, Long idDisciplinaUnifica) {

        this.id = id;
        this.codigo = exigirNaoNulo(codigo, "Código");
        this.nome = exigirTamanho(nome, "Nome", 1, 100);
        this.sigla = exigirTamanho(sigla, "Sigla", 1, 20);  
        this.cargaHorariaTotal = exigirNaoNulo(cargaHorariaTotal, "Carga horária total");
        this.cargaHorariaPresencial = exigirNaoNulo(cargaHorariaPresencial, "Carga horária presencial"); 
        this.cargaHorariaDistancia = exigirNaoNulo(cargaHorariaDistancia, "Carga horária à distância");
        this.ativa = exigirNaoNulo(ativa, "Ativa"); 
        this.obrigatoria = exigirNaoNulo(obrigatoria, "Obrigatória");
        this.valor = exigirNaoNulo(valor, "Valor"); 
        this.idTipoDisciplina = exigirNaoNulo(idTipoDisciplina, "Tipo da disciplina");
        this.idTipoRegimeLetivo = exigirNaoNulo(idTipoRegimeLetivo, "Tipo do regime letivo"); 
        this.idPeriodo = exigirNaoNulo(idPeriodo, "Período");
        this.idComponente = idComponente;
        this.dataInicioVigencia = exigirNaoNulo(dataInicioVigencia, "Data de início da vigência"); 
        this.dataFinalVigencia = dataFinalVigencia; 
        this.preRequisitos = exigirNaoNulo(preRequisitos, "Pré-requisitos");
        this.idDisciplinaUnifica = idDisciplinaUnifica;
	}
}
